core/geometry/src/polyface/PolyfaceQuery.ts

```diff
@@ -321,66 +318,65 @@ export class PolyfaceQuery {
-            if (clipFractions.clipBy01FunctionValuesPositive(localSegmentPoint0.x, localSegmentPoint1.x)
-              && clipFractions.clipBy01FunctionValuesPositive(localSegmentPoint0.y, localSegmentPoint1.y)
-              && clipFractions.clipBy01FunctionValuesPositive(
-                1 - localSegmentPoint0.x - localSegmentPoint0.y,
-                1 - localSegmentPoint1.x - localSegmentPoint1.y)) {
-              /* project the local segment point to the plane. */
-              const localClippedPointA = localSegmentPoint0.interpolate(clipFractions.x0, localSegmentPoint1);
-              const localClippedPointB = localSegmentPoint0.interpolate(clipFractions.x1, localSegmentPoint1);
-              const worldClippedPointA = localFrame.multiplyPoint3d(localClippedPointA)!;
-              const worldClippedPointB = localFrame.multiplyPoint3d(localClippedPointB)!;
-              const planePointA = localFrame.multiplyXYZ(localClippedPointA.x, localClippedPointA.y, 0.0)!;
-              const planePointB = localFrame.multiplyXYZ(localClippedPointB.x, localClippedPointB.y, 0.0)!;
-              const splitParameter = Geometry.inverseInterpolate01(localSegmentPoint0.z, localSegmentPoint1.z);
-              // emit 1 or 2 panels, oriented so panel normal is always to the left of the line.
-              if (splitParameter !== undefined && splitParameter > clipFractions.x0 && splitParameter < clipFractions.x1) {
-                const piercePointX = segmentPoint0.interpolate(splitParameter, segmentPoint1);
-                const piercePointY = piercePointX.clone();   // so points are distinct for the two triangle announcements.
-                announce(linestringPoints, i1 - 1, polyface, visitor.currentReadIndex(), [worldClippedPointA, piercePointX, planePointA], 2, 1);
-                announce(linestringPoints, i1 - 1, polyface, visitor.currentReadIndex(), [worldClippedPointB, piercePointY, planePointB], 1, 2);
-              } else if (localSegmentPoint0.z > 0) {  // segment is entirely above
-                announce(linestringPoints, i1 - 1, polyface, visitor.currentReadIndex(), [worldClippedPointA, worldClippedPointB, planePointB, planePointA], 3, 2);
-              } else // segment is entirely under
-                announce(linestringPoints, i1 - 1, polyface, visitor.currentReadIndex(), [worldClippedPointB, worldClippedPointA, planePointA, planePointB], 2, 3);
-            }
-          }
-        }
+    const context = SweepLineStringToFacetContext.create(linestringPoints);
+    if (context) {
+      const visitor = polyface.createVisitor(0);
+      for (visitor.reset(); visitor.moveToNextFacet();) {
+        context.projectToPolygon(visitor.point, announce, polyface, visitor.currentReadIndex());
+      }
+    }
+  }
+
+
+  /** Execute context.projectToPolygon until its work estimates accumulate to workLimit  */
+  private static async continueAnnouunceSweepLinestringToConvexPolyfaceXY(

```

Comment:
typo, "Annouunce"

Reply:
Ah, uuuuuuuuuuuu... it's private, so I'll fix on next PR.